executor/executor.go

```diff
@@ -1975,7 +1975,7 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 	if _, ok := s.(*ast.AnalyzeTableStmt); ok {
 		sc.InitMemTracker(memory.LabelForAnalyzeMemory, -1)
 		vars.MemTracker.SetBytesLimit(-1)
-		vars.MemTracker.AttachTo(GlobalAnalyzeMemoryTracker)
+		sc.MemTracker.AttachTo(GlobalAnalyzeMemoryTracker)
```

Comment:
The normal tracker chain is `stmt tracker -> session tracker -> global tracker`. Here we change it to `stmt tracker -> global analyze tracker -> global tracker`. I am not sure whether it has some unexpected impact. cc @XuHuaiyu

Reply:
We must attach stmt tracker to session tracker, so that global-mem-limit can see manual analyze as top1 consumer from session manager. 
While session manager cannot see auto-analyze which is run from a background shared session pool, will handle it in another PR.

Reply:
https://github.com/pingcap/tidb/pull/39978